File: 3723-sum-of-good-numbers/sum-of-good-numbers.cpp
class Solution {
public:
    int sumOfGoodNumbers(vector<int>& nums, int k) {
        int n = nums.size();
        int goodSum = 0;

        for (int i = 0; i < n; ++i) {
            if (i - k >= 0 && nums[i] <= nums[i - k]) {
                continue;
            }
            if (i + k < n && nums[i] <= nums[i + k]) {
                continue;
            }
            goodSum += nums[i];
        }

        return goodSum;
    }
};

File: 3723-sum-of-good-numbers/sum-of-good-numbers.py
class Solution:
    def sumOfGoodNumbers(self, nums: list[int], k: int) -> int:
        n = len(nums)
        Gsum = 0

        for i, num in enumerate(nums):
            if i - k >= 0 and num <= nums[i - k]:
                continue
            if i + k < n and num <= nums[i + k]:
                continue
            Gsum += num

        return Gsum
